### DISCUSSION TITLE
Drawing circle sectors?

### DISCUSSION THREAD

**Discussion by ghost** [cduck/drawSvg#127] [Q&A]
Hello, I really appreciate your library, it has been useful *and* easy to use for me so far :)

I couldn't find a reference to drawing circle sectors in your documentation, is there a way to do it? In my case they have arbitrary angles, not just 90°s (which could be done using a rectangle clip like in the clip demo in the tutorial)

![c35a90e3f084d9531156d2c292015b24](https://github.com/cduck/drawsvg/assets/11392207/2d94c30c-12a8-44ec-819e-66ec6dad4278)

**Comment by cduck** [comment]
I'm glad it's useful for you!  There is no specific element for arc segments so you'll need to manually use the `Path` element with an arc and two lines.  And you'll need to do some trigonometry (like `math.sin(radian_angle)` in Python) to calculate the start and end points of the arc relative to the center.  See https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#arcs (the very last example) for how to use SVG arc paths.  Good luck!

**Comment by NATateo** [comment]
I successfully created a sector by drawing an arc with large stroke_width and no fill, you only have to fiddle with the proper parameters (radius is half the radius of the circle).
It's limited cause you can't have an outline, since it literally is just one big stroke, but for my case it worked.

**Comment by ghost** [comment]
Following [this](https://stackoverflow.com/a/24569190) StackOverflow answer, I got together this code, which appears to work :)

Edit: this behaves weirdly if the start_angle and end_angle are the same (0° or 360° circle)

```python
from math import sin, cos, pi

import drawsvg as dw

def polar_to_cartesian(cx, cy, r, a):
	a_rad = a*pi/180.0
	x = cx + r * cos(a_rad)
	y = cy + r * sin(a_rad)
	return x,y

# angles are given in degrees
def circle_sector(d, x, y, radius, start_angle, end_angle, stroke="red", fill="black"):
	start = polar_to_cartesian(x, y, radius, end_angle)
	end = polar_to_cartesian(x, y, radius, start_angle)
	
	arc_sweep = 0 if end_angle - start_angle <= 180 else 1
	
	p = dw.Path(stroke=stroke, fill=fill)
	p.M(start[0], start[1])
	p.A(radius, radius, rot=0, large_arc=0, sweep=arc_sweep, ex=end[0], ey=end[1])
	p.L(x,y)
	p.L(start[0], start[1])
	
	d.append(p)

width = height = 512
d = dw.Drawing(width, height, origin=(-width//2,-height//2))

for start_angle in range(0, 360, 10):
	circle_sector(d, 0, 0, 200, start_angle, start_angle+5, "red", "black")
	
d.save_svg("sector.svg")
```
![sector](https://github.com/cduck/drawsvg/assets/11392207/c0b08aab-5eb7-4489-aa13-da4d3b457b94)


It would be cool if this functionality would be included in the library for future users
